cl/citations/match_citations.py

```diff
@@ -120,7 +128,7 @@ def match_citation(citation, citing_doc=None):
     }
     if citing_doc is not None:
         # Eliminate self-cites.
-        main_params["fq"].append("-id:%s" % citing_doc.pk)
+        cast(List[str], main_params["fq"]).append("-id:%s" % citing_doc.pk)
```

Comment:
This `cast()` call is a bit ugly but I found it necessary because mypy doesn't know whether this particular parameter (`fq`) maps to a str, int, or a list of strings.

Reply:
(ditto for below)

Reply:
`SearchParam` will fix this too.